synapse/module_api/__init__.py

```diff
@@ -73,6 +87,62 @@ def public_room_list_manager(self):
         """
         return self._public_room_list_manager
 
+    def register_hook(self, fn_name: str, callback: Callable):
+        """Registers the given callback as a hook with the given name.
+
+        This function must be called in the __init__ method of the modules using it.
```

Comment:
It feels like we might want to error if this function isn't called in the `__init__` method of a module, but I'm not sure how to check that nor whether it's worth the pain.

Reply:
I don't think it's necessary tbh. Why must it be called in the __init__ method specifically? It seems the obvious way, but ultimately it's up for the module to ensure that its hooks are registered during startup. If it wants to do it some other way, that's ok by us.